code-challenges.test.js

```diff
@@ -11,49 +11,198 @@
+    expect(divisibleByThree(num2)).toEqual(`0 is divisible by three`)
+  })
+})
+
 var num3 = -7
+console.log(divisibleByThree(num3))
 // Expected output: "-7 is not divisible by three"
 
+describe("divisibleByThree", () => {
+  it("returns a string of `-7 is not divisible by three` if a number is divisible by 3 or not based on a number input", () => {
+    expect(divisibleByThree(num3)).toEqual(`-7 is not divisible by three`)
+  })
+})
 
 
 // b) Create the function that makes the test pass.
 
+// //Refactored function copied to line 35-40
+// const divisibleByThree = (num) => {
+//   if (num % 3 === 0) {
+//     return `${num} is divisible by three`;
+//     //refactored the code to be slightly more concise by adding the ";" to stop the function when the if condition returns true.
+//   }
+//   return `${num} is not divisible by three`
+//   //An else statememt is not needed when using the ";" operator/symbol in the preceding "if" condition. Whenever the if condition is not met, the function will run to the next line of code. 
+// }
+// //Refactored function copied to line 35-40
 
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
+//Declare a function that iterates over an array
+const capitalPunishment = (array) => {
+  //Use the .map higher order function to apply an operation over every value in the array.
+  return array.map(value => value.charAt(0).toUpperCase() + value.slice(1))
+  //The first index of each string on the array will be the first letter, to change the first letter I used the .charAt(0) and applied the .toUppercase methods to capitalize the first letter. Adding the capitalised letter to the same string I used the .slice method to create a string missing the first index/first letter. .slice(1) on a string will create a string missing the first letter/index. 
+};
+
+
 // a) Create a test with expect statements for each of the variables provided.
 
 var randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+console.log(capitalPunishment(randomNouns1));
+
+describe("capitalPunishment", () => {
+  it("returns a new array with all string elements capitalized", () => {
+    expect(capitalPunishment(randomNouns1)).toContain( "Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew" )
+  })
+})
+
 var randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
+console.log(capitalPunishment(randomNouns2));
 
 
+describe("capitalPunishment", () => {
+  it("returns a new array with all string elements capitalized", () => {
+    expect(capitalPunishment(randomNouns2)).toContain( "Temperature", "Database", "Chopsticks", "Mango", "Deduction" )
+  })
+})
 
 // b) Create the function that makes the test pass.
```

Comment:
Just a heads keep your functions under the B section to give me the illusion (or reality) of writing them after your tests.